lib/pf/pfcmd/checkup.pm

```diff
@@ -1098,6 +1103,52 @@ sub db_check_version {
     }
 }
 
+=item valid_certs
+
+Make sure the certificates used by Apache and RADIUS are valid
+
+=cut
+
+sub valid_certs {
+    my $httpd_conf = read_file("$generated_conf_dir/ssl-certificates.conf");
+
+    my ($httpd_crt, $radius_crt);
+
+    if($httpd_conf =~ /SSLCertificateFile\s*(.*)/){
+        $httpd_crt = $1;
+    }
+    else{
+        add_problem($FATAL, "Cannot find the Apache certificate in your configuration.");
+    }
+
+    my $radius_conf = read_file("$install_dir/raddb/eap.conf");
+
+    if($radius_conf =~ /certificate_file =\s*(.*)/){
+        $radius_crt = $1;
+    }
+    else{
+        add_problem($FATAL, "Cannot find the FreeRADIUS certificate in your configuration.");
+    }
+
+
+    my ($expiration, $cert);
+
+    $cert = Crypt::OpenSSL::X509->new_from_file($httpd_crt);
+
+    $expiration = str2time($cert->notAfter);
+    if(time > $expiration){
+        add_problem($FATAL, "The certificate used by Apache ($httpd_crt) has expired.\nRegenerate a new self-signed certificate or update your current certificate.");
+    }
+
+    $cert = Crypt::OpenSSL::X509->new_from_file($radius_crt);
+
+    $expiration = str2time($cert->notAfter);
+    if(time > $expiration){
+        add_problem($FATAL, "The certificate used by FreeRADIUS ($radius_crt) has expired.\nRegenerate a new self-signed certificate or update your current certificate.");
+    }
```

Comment:
Look good could you refactor check in a function.
So it would be easier to add additional cert files in the future
`
sub _cert_file_has_expired {
    $cert = Crypt::OpenSSL::X509->new_from_file($httpd_crt);
    $expiration = str2time($cert->notAfter);
    return time > $expiration;
}

`